crates/storage/src/db.rs

```diff
@@ -190,4 +188,46 @@ impl ReamDB {
             db: self.db.clone(),
         }
     }
+
+    pub fn is_initialized(&self) -> bool {
+        match self.slot_index_provider().get_highest_slot() {
+            Ok(Some(slot)) => slot > 0,
+            _ => false,
+        }
+    }
+}
+
+pub fn reset_db(db_path: PathBuf) -> anyhow::Result<()> {
+    if fs::read_dir(&db_path)?.next().is_none() {
+        info!("Data directory at {db_path:?} is already empty.");
+        return Ok(());
+    }
+
+    info!(
+        "Are you sure you want to clear the contents of the data directory at {db_path:?}? (y/n):"
+    );
+    let mut input = String::new();
+    io::stdin().read_line(&mut input)?;
+    if input.trim().eq_ignore_ascii_case("y") {
+        for entry in fs::read_dir(&db_path)? {
+            let entry = entry?;
+            let path = entry.path();
+            if path.is_dir() {
+                fs::remove_dir_all(&path)?;
+            } else {
+                fs::remove_file(&path)?;
+            }
+        }
+        info!("Database contents cleared successfully.");
+    } else {
+        info!("Operation canceled by user.");
+    }
+    Ok(())
+}
+
+pub fn initialize_db_dir(
+    data_dir: Option<PathBuf>,
+    ephemeral: bool,
+) -> Result<PathBuf, StoreError> {
+    dir::setup_data_dir(APP_NAME, data_dir, ephemeral).map_err(StoreError::Io)
```

Comment:
I think we should just use setup_data_dir directly, no need for an abstraction for 1 line